sceptre/config/graph.py

```diff
@@ -2,87 +2,113 @@
+
+    def remove_stack(self, stack):
         """
-        Generates the graph for the initalized StackDependencyGraph object
+        Removes a Stack from the StackGraph. This operation will also remove
+        all adjecent edges that represent a 'depends on' relationship with
+        other Stacks.
         """
-        for stack, dependencies in self.dependency_map.items():
-            self._generate_edges(stack, dependencies)
+        return self.graph.remove_node(stack)
 
-    def _is_acyclic(self):
-        return nx.is_directed_acyclic_graph(self.graph)
+    def write(self):
+        nx.drawing.nx_pydot.write_dot(
+            self.graph,
+            './out.dot'
+        )
 
-    def _generate_edges(self, stack_path, dependency_paths):
+        print(list(reversed(
+            nx.algorithms.dag.dag_longest_path(self.graph)
+        )))
+
+        print(nx.convert.to_dict_of_lists(self.graph))
+
+    def _generate_graph(self, stacks):
+        """
+        Generates the graph for the StackGraph object.
+
+        :param stacks: A set of Stacks
+        :type stacks: set
+        """
+
+        for stack in stacks:
+            self._generate_edges(stack, stack.dependencies)
+        self.graph.remove_edges_from(nx.selfloop_edges(self.graph))
+
+    def _generate_edges(self, stack, dependencies):
         """
         Adds edges to the graph based on a list of dependencies that are
         generated from the inital stack config. Each of the paths
         in the inital_dependency_paths list are a depency that the inital
         Stack config depends on.
 
-        :param dependency_paths: a collection of dependency paths
-        :type inital_dependency_paths: string
+        :param stack: A Sceptre Stack
+        :type stack: sceptre.stack.Stack
+        :param dependencies: a collection of dependency paths
+        :type dependencies: list
         """
         self.logger.debug(
             "Generate edges for graph {0}".format(self.graph)
         )
-        for dependency_path in dependency_paths:
-            edge = self.graph.add_edge(stack_path, dependency_path)
-            if not self._is_acyclic():
+        for dependency in dependencies:
+            edge = self.graph.add_edge(dependency, stack)
+            if not nx.is_directed_acyclic_graph(self.graph):
                 raise CircularDependenciesError(
-                    "Dependency cycle detected: {} {}".format(stack_path,
-                                                              dependency_path))
+                    "Dependency cycle detected: {} {}".format(stack,
+                                                              dependency))
             self.logger.debug("Added edge: {}".format(edge))
 
-        if not dependency_paths:
-            self.graph.add_node(stack_path)
-
-        self.graph.remove_edges_from(nx.selfloop_edges(self.graph))
+        if not dependencies:
+            self.graph.add_node(stack)
 
     def reverse_graph(self):
```

Comment:
I think we can remove `reverse_graph()` as it does not appear to be called anywhere?